index.md

```diff
@@ -1,16 +1,128 @@
+
+The third is the most difficult.  In Nublado, we have a spawner form.  That form contains a selection of prepulled images, a drop-down box to select an uncached image, a selector to choose container size, and two checkboxes useful for debugging and unwedging stuck environments.  I am not actually sure how generic that is, but I think erring on the side of "everyone will want something like this" is better than insisting it's DM-pipelines-specific.
+
+The purpose of the form in the RSP environment is basically twofold: first, to let the user select how much they need in the way of resources, and second to reflect our release cadence.  We have, effectively, "daily", "weekly", and "release" images and would like some way to present those to the user and update them without human intervention as new images are built.
+
+By having rules about how many of each we use, we can also decide which images to prepull.  Given our use cases, it is vital to prepull images, because they are so large that waiting for a spawn that needs to pull an image from a remote repository becomes quite an unpleasant experience.  If the images are cached to each node (which Nublado does), then for images displayed on the menu (rather than in the drop-down list of all images), the spawn is generally a matter of seconds rather than minutes.  We simply poll the image repository and pull the appropriately-tagged images periodically.  Most of the time, neither the tags nor the content at each tag has changed, and in this case, all the "pull" is the server replying that the image associated with a tag has a particular sha256 checksum, and the client saying "oh, I've got that one, nothing to do, then"; thus while we do many thousands of pulls a day, only a handful actually do much work.
+
+It is not entirely clear that the Rubin practice of builds-on-a-fairly-short-cadence, almost immediately reflected in the spawner, is what all users of Phalanx will want.  However, this form would work just as well with a single type of release and builds updated quarterly, although the mechanisms that populate it will be overcomplicated for the use case.
+
+Assuming the spawner options form is fundamentally generic, then there needs to be some method of categorizing and sorting images.  Currently this is the `RSPTag` class, buried deep inside the Lab controller, which understands the tag conventions used by the DM pipelines container builds.  This needs to be somehow broken out of the controller, and it needs to be easy to replace `RSPTag` with some other class that recognizes image tags in whatever format the payload repository owner uses.  That might be as simple as semantic or date-based versioning.  Whatever it is will need to have some way to map a tag schema to categorization of image type and a sort order within each type, in order to construct an appropriate spawner form and to direct prepulling.
```

Comment:
It's not immediately obvious to me why `RSPTag` needs to be broken out of the controller. Presumably you have some use case in mind. It might be helpful to mention it explicitly.